numba/tests/test_typedlist.py

```diff
@@ -701,3 +701,42 @@ def foo():
         got = foo()
         # Need to compare the nested arrays
         self.assertTrue(np.all(expected[0] == got[0]))
+
+    @skip_py2
+    def test_jitclass_as_item_in_list(self):
+
+        spec = [
+            ('value', int32),               # a simple scalar field
+            ('array', float32[:]),          # an array field
+        ]
+
+        @jitclass(spec)
+        class Bag(object):
+            def __init__(self, value):
+                self.value = value
+                self.array = np.zeros(value, dtype=np.float32)
+
+            @property
+            def size(self):
+                return self.array.size
+
+            def increment(self, val):
+                for i in range(self.size):
+                    self.array[i] += val
+                return self.array
+
+        @njit
+        def foo():
+            l = List()
+            l.append(Bag(21))
+            l.append(Bag(22))
+            l.append(Bag(23))
+            return l
+
+        expected = foo.py_func()
+        got = foo()
+
+        def bag_equal(one, two):
+            # jitclasses couldn't override __eq__ at time of writing
+            return one.value == two.value and all(one.array == two.array)
+        self.assertTrue(all([bag_equal(a,b) for a,b in zip(expected, got)]))
```

Comment:
How about this, such that if there is a problem, it will be reported at the site and with the erroneous values: 
```python
        def bag_equal(one, two):
            # jitclasses couldn't override __eq__ at time of writing
            self.assertEqual(one.value, two.value)
            np.testing.assert_allclose(one.array, two.array)

        [bag_equal(a,b) for a, b in zip(expected, got)]
```
?

Reply:
Sure.